stages/02_download.sh

```diff
@@ -88,8 +84,57 @@ for data_type in $data_types; do
   # Get the list of field names (e.g., enforcement, event, 510k) for each data type
   field_names=$(jq -r ".results.\"${data_type}\" | keys[]" "$listjson")
 
-  # Iterate through each field name and call the download function
+  # Iterate through each field name
   for field_name in $field_names; do
-    download_files "$data_type" "$field_name"
+    # Get the number of partitions
+    num_partitions=$(jq -r ".results.\"${data_type}\".\"${field_name}\".partitions | length" "$listjson")
+    
+    # Add each partition to the download list as tab-separated values
+    for (( i=0; i<num_partitions; i++ )); do
+      echo -e "${data_type}\t${field_name}\t${i}\t${listjson}" >> "$download_list_file"
+    done
   done
 done
+
+num_files=$(wc -l < "$download_list_file")
+echo "Found $num_files files to download"
+
+# Use GNU parallel to download files in parallel with a single clean progress bar
+if [[ $num_files -gt 0 ]]; then
+  echo "Starting parallel downloads..."
+  
+  # Create a custom progress function
+  progress_file=$(mktemp)
+  echo "0" > "$progress_file"
+  
+  # Modified download function that updates progress
+  download_with_progress() {
```

Comment:
Instead of this, use `--bar` on the `parallel` command and have that call `download_single_file`.